core/src/main/java/org/infinispan/statetransfer/StateTransferLockImpl.java

```diff
@@ -96,6 +96,7 @@ public void notifyTransactionDataReceived(int topologyId) {
          if (transactionDataTopologyId >= expectedTopologyId) {
             return CompletableFutures.completedNull();
          } else {
+            // TODO Dan: Use thenComposeAsync to continue each command in a separate thread?
```

Comment:
No, the thread in which the execution should continue should be decided by the caller of `transactionDataFuture`.

Reply:
Good point, so we just need to modify `DefaultTopologyRunnable` to use `thenApplyAsync`... or remove `DefaultTopologyRunnable` altogether :)

Reply:
`BaseStateTransferInterceptor` also uses `transactionDataReceived()` directly, so I created https://issues.jboss.org/browse/ISPN-9852